sharding/nodesCoordinator/indexHashedNodesCoordinator_test.go

```diff
@@ -1305,6 +1323,36 @@ func TestIndexHashedNodesCoordinator_EpochStartInEligible(t *testing.T) {
 	require.True(t, isValidator)
 }
 
+func TestIndexHashedNodesCoordinator_computeShardForSelfPublicKeyWithStakingV4(t *testing.T) {
+	t.Parallel()
+
+	arguments := createArguments()
+	pk := []byte("pk")
+	arguments.SelfPublicKey = pk
+	nc, _ := NewIndexHashedNodesCoordinator(arguments)
+	epoch := uint32(2)
+
+	metaShard := core.MetachainShardId
+	nc.nodesConfig = map[uint32]*epochNodesConfig{
+		epoch: {
+			shardID: metaShard,
+			shuffledOutMap: map[uint32][]Validator{
+				metaShard: {newValidatorMock(pk, 1, 1)},
+			},
+		},
+	}
+
+	computedShardId, isValidator := nc.computeShardForSelfPublicKey(nc.nodesConfig[epoch])
+	require.Equal(t, nc.shardIDAsObserver, computedShardId)
+	require.False(t, isValidator)
+
+	nc.flagStakingV4.SetReturningPrevious()
```

Comment:
why not SetValue(true) as you are not checking the returned value anyway?

Reply:
`SetReturningPrevious` was doing the same thing, but I guess it's better to use `SetValue`. Refactored as suggested